### PR TITLE
feat(typescript): add bcdb

### PR DESCRIPTION
**Pull request by bearmit**
This pull request is related to the card
https://trello.com/c/ZTRYwptP/48-cript-scripts-bcdb

The source data and the generated JSON can be found here https://drive.google.com/drive/folders/17AG2XVLixOB2XuDezbc9CdPiFQ7nLl37

# What does the script?

1. Read the blocks tab from the bcdb.xlsx (to have a polymer/name/bigsmiles correspondence)
2. Read each line of the diblock tab from bcbd.xlsx
For each line, we:
- create a citation (including a reference)
- create a main material (polymer)
- we add 2 components on it (block 1 and 2).

Many properties/conditions are also added, see the source data (bcdb.column_meaning.xls) for more info.

### CODE REVIEW

File: scripts/typescript/src/bcdb/README.md
Comment (bearmit) [inline scripts/typescript/src/bcdb/README.md]:
```suggestion
This script is able to convert an xlsx file to CRIPT JSON.
```

File: scripts/typescript/src/bcdb/bcdb.ts
Comment (bearmit) [inline scripts/typescript/src/bcdb/bcdb.ts]:
```suggestion
 * The purpose of this script is to load  bdcb.xlsx file and export it as a JSON file.
 *
```

File: scripts/typescript/src/bcdb/bcdb.ts
Comment (bearmit) [inline scripts/typescript/src/bcdb/bcdb.ts]:
```suggestion
```

File: scripts/typescript/src/bcdb/bcdb.ts
Comment (bearmit) [inline scripts/typescript/src/bcdb/bcdb.ts]:
```suggestion
 * This script does not import all the sheets and columns.
```

File: scripts/typescript/src/bcdb/bcdb.ts
Comment (bearmit) [inline scripts/typescript/src/bcdb/bcdb.ts:8]: I think I should include the column_meaning.xlsx to the repo, since there is the xlsx for the data too. I cannot reference the drive (it won't be accessible by the public, and we don't want it).

File: scripts/typescript/src/bcdb/bcdb.ts
Comment (bearmit) [inline scripts/typescript/src/bcdb/bcdb.ts]:
```suggestion
    log(`=-=-=-=-=-=-=-=-=-= BCDB XLSX to JSON =-=-=-=-=-=-=-=-=`);
```

File: scripts/typescript/src/bcdb/bcdb.ts
Comment (bearmit) [inline scripts/typescript/src/bcdb/bcdb.ts]:
```suggestion
```

File: scripts/typescript/src/bcdb/bcdb.ts
Comment (bearmit) [inline scripts/typescript/src/bcdb/bcdb.ts:112]: I am not sure which one to pick, because it is an algorithm but I am citing a paper (doi with ORCID).

>  "book",
                    "journal_article",
                    "encyclopedia_article",
                    "news_paper",
                    "patent",
                    "thesis",
                    "web_site",
                    "database"

File: scripts/typescript/src/bcdb/bcdb.ts
Comment (bearmit) [inline scripts/typescript/src/bcdb/bcdb.ts:189]: `rheology` is not available in the vocab. We have to add it or pick a different one.

> "prescribed",
                            "comp",
                            "nmr",
                            "sec",
                            "gc",
                            "chrom",
                            "ms",
                            "maldi",
                            "ultra_centr",
                            "osmtic_pres",
                            "calorimetry",
                            "cryoscopy",
                            "ebullioscopy",
                            "viscometer",
                            "utm",
                            "dma",
                            "dsc",
                            "tga",
                            "raman",
                            "ir",
                            "near-ir",
                            "uv_vis",
                            "x_ray",
                            "saxs",
                            "waxs",
                            "neutron",
                            "ls",
                            "dls",
                            "confocal",
                            "afm",
                            "tem",
                            "sem",
                            "scale",
                            "mals",
                            "membrane_osmometry",
                            "sec_mals",
                            "sec_visco",
                            "sec_ir",
                            "ASTM_D3574_Test_A",
                            "ASTM_D3574_Test_B1",
                            "ASTM_D3574_Test_B2",
                            "ASTM_D3574_Test_C",
                            "ASTM_D3574_Test_D",
                            "ASTM_D3574_Test_E",
                            "ASTM_D3574_Test_F",
                            "ASTM_D3574_Test_G",
                            "ASTM_D3574_Test_H",
                            "ASTM_D3574_Test_I1",
                            "ASTM_D3574_Test_I2",
                            "ASTM_D3574_Test_I3",
                            "ASTM_D3574_Test_I4",
                            "ASTM_D3574_Test_I5",
                            "ASTM_D3574_Test_J",
                            "ASTM_D3574_Test_K",
                            "ASTM_D3574_Test_L",
                            "ASTM_D3574_Test_M"

File: scripts/typescript/src/bcdb/bcdb.ts
Comment (bearmit) [inline scripts/typescript/src/bcdb/bcdb.ts:339]:
```suggestion
            // try to get meta data from blocks map
            // meta data are coming from the 'blocks' tab in the xlsx
```

File: scripts/typescript/src/bcdb/index.ts
Comment (bearmit) [inline scripts/typescript/src/bcdb/index.ts]:
```suggestion
 * The purpose of this script is to convert BCBD's XLSX to CRIPT JSON.
 * @see BCBDLoader for more information
 */
```

**Comment by nh916** [comment]
I just ran the Jekyll website locally and it seems okay. 

I don't know the contents of the script

@bearmit when you got to merge, I recommend squash and merge to keep master branch commits clean

**Review by nh916 — APPROVED** [review]

File: scripts/typescript/src/bcdb/bcdb.ts
Comment (bearmit) [inline scripts/typescript/src/bcdb/bcdb.ts:128]: Same remark, I picked `reference` but I clearly don't know if it is the correct value.

> "extracted_by_human",
                    "extracted_by_nlp",
                    "extracted_by_algorithm", // could be this too
                    "derived_from",
                    "replicated",
                    "reference"

**Comment by bearmit** [comment]
@dylan , @njrebello: Do you think you will be able to take a look at the code? I would like to merge this branch not too late, if not I'll completely forgot what and how I did it. Anyways just let me know, If you don't I'll merge the branch as-is.
Thanks.

**Comment by nh916** [comment]
> @dylan , @njrebello: Do you think you will be able to take a look at the code? I would like to merge this branch not too late, if not I'll completely forgot what and how I did it. Anyways just let me know, If you don't I'll merge the branch as-is. Thanks.

I strongly recommend comments and documentation if needed/feasible because that has helped me in the past when switching between code and to be able to read the code after a while of not seeing it

**Comment by bearmit** [comment]
> I strongly recommend comments and documentation if needed/feasible because that has helped me in the past when switching between code and to be able to read the code after a while of not seeing it

You're right, I just double checked and I think comments are sufficient to understand.

That said, this script is not something that we will maintained I believe, it smells like "thrash able" code to me. I did this in TS because of the lack of a SDK and the need to prepare some test data the fastest we can. 
It is not very well structured because of that and we should rely on a SDK next time.
.
@dylanwal or @njrebello can you answer to my questions above?

File: scripts/typescript/src/bcdb/bcdb.ts
Comment (olsenlabmit) [reply scripts/typescript/src/bcdb/bcdb.ts:128]: extracted_by_human

File: scripts/typescript/src/bcdb/bcdb.ts
Comment (olsenlabmit) [reply scripts/typescript/src/bcdb/bcdb.ts:112]: journal_article

**Comment by nh916** [comment]
@bearmit would it be okay if I delete this branch, or are you still using it?

**Comment by bearmit** [comment]
> @bearmit would it be okay if I delete this branch, or are you still using it?

 I prefer to keep it until the other is merged too.

**Pull request merged**
